zookeeper-server/src/main/java/org/apache/zookeeper/server/ZKDatabase.java

```diff
@@ -314,26 +314,24 @@ private void addCommittedProposal(TxnHeader hdr, Record txn, TxnDigest digest) {
      * fast follower synchronization.
      * @param request committed request
      */
-    public void addCommittedProposal(Request request) {
+    public void addCommittedProposal(final Request request) {
         WriteLock wl = logLock.writeLock();
+        wl.lock();
         try {
-            wl.lock();
-            if (committedLog.size() > commitLogCount) {
-                committedLog.remove();
-                minCommittedLog = committedLog.peek().packet.getZxid();
-            }
-            if (committedLog.isEmpty()) {
-                minCommittedLog = request.zxid;
-                maxCommittedLog = request.zxid;
-            }
-
             byte[] data = SerializeUtils.serializeRequest(request);
             QuorumPacket pp = new QuorumPacket(Leader.PROPOSAL, request.zxid, data, null);
             Proposal p = new Proposal();
             p.packet = pp;
             p.request = request;
+
+            if (committedLog.size() == commitLogCount) {
```

Comment:
Interesting.  I was initially puzzled by the change from `>`  to `==`. This also fixes an off-by-one error in the existing code (and, consequently, makes the log one entry shorter), doesn't it?  I'd probably add a bullet point saying so to the description.

Reply:
@ztzg Lol.  I actually didn't even realize I fixed that, but you're correct.  I'll update the docs.